src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc

```diff
@@ -235,6 +238,57 @@ GcsPlacementGroupScheduler::GetOrConnectLeaseClient(const rpc::Address &raylet_a
   return iter->second;
 }
 
+void GcsPlacementGroupScheduler::OnPlacementGroupResourceReserveDone(
+    const std::shared_ptr<GcsPlacementGroup> &placement_group,
+    const std::vector<std::shared_ptr<BundleSpecification>> &bundles,
+    const std::shared_ptr<BundleLocation> &bundle_locations,
+    const std::function<void(std::shared_ptr<GcsPlacementGroup>)>
+        &schedule_failure_handler,
+    const std::function<void(std::shared_ptr<GcsPlacementGroup>)>
+        &schedule_success_handler) {
+  const auto &placement_group_id = placement_group->GetPlacementGroupID();
+  if (placement_group_leasing_in_progress_.find(placement_group_id) ==
+          placement_group_leasing_in_progress_.end() ||
+      bundle_locations->size() != bundles.size()) {
+    // If the lease request has been already cancelled
+    // or not every lease request succeeds,
+    for (auto &iter : *bundle_locations) {
+      auto &bundle_spec = iter.second.second;
+      auto &node_id = iter.second.first;
+      CancelResourceReserve(bundle_spec, gcs_node_manager_.GetNode(node_id));
```

Comment:
placement_group_to_bundle_location_.erase(...) ?

Reply:
Good catch. To make it more clean, I emplaced the placement_group_to_bundle_location_ in the beginning of the method, and just call `DestroyPlacementGroupResources `, which removes an entry from `placement_group_to_bundle_location_`